drv/lpc55-update-server/src/images.rs

```diff
@@ -26,79 +45,494 @@ extern "C" {
 // Location of the NXP header
 pub const HEADER_BLOCK: usize = 0;
 
-// NXP LPC55's mixed header/vector table offsets
-const RESET_VECTOR_OFFSET: usize = 0x04;
-pub const LENGTH_OFFSET: usize = 0x20;
-pub const HEADER_OFFSET: u32 = 0x130;
-const MAGIC_OFFSET: usize = HEADER_OFFSET as usize;
+// An image may have an ImageHeader located after the
+// LPC55's mixed header/vector table.
+pub const IMAGE_HEADER_OFFSET: u32 = 0x130;
 
-// Perform some sanity checking on the header block.
-pub fn validate_header_block(
-    component: RotComponent,
-    slot: SlotId,
-    block: &[u8; BLOCK_SIZE_BYTES],
-) -> Result<(), UpdateError> {
-    let exec = image_range(component, slot).1;
+/// Address ranges that may contain an image during storage and active use.
+/// `store` and `exec` ranges are the same except for `stage0next`.
+pub struct FlashRange {
+    pub store: Range<u32>,
```

Comment:
Not a problem in our current address space layout, but in general it seems like these should be RangeInclusive --- otherwise they can't model a range at the end of the u32 space.

Reply:
Added a TODO for now. I'll come back to this.

Reply:
https://github.com/oxidecomputer/hubris/issues/1836